mmcv/runner/base_runner.py

```diff
@@ -195,14 +197,70 @@ def val(self):
     def run(self, data_loaders, workflow, **kwargs):
         pass
 
-    @abstractmethod
     def save_checkpoint(self,
                         out_dir,
-                        filename_tmpl,
+                        filename_tmpl='epoch_{}.pth',
```

Comment:
Using `epoch_{}.pth` as the default checkpoint name for `iter_based_runner` may be a little confusing for users. A possible method is using None as default, but getting changed to `iter_{}.pth` or `epoch_{}.pth` according to `by_epoch` if it is None. But I am not sure if it is the best solution here.